File: asset_management/page/maintenance_dashboard/maintenance_dashboard.js
frappe.pages['maintenance-dashboard'].on_page_load = function(wrapper) {
    const page = frappe.ui.make_app_page({
        parent: wrapper,
        title: 'Maintenance Dashboard',
        single_column: true
    });

    $(frappe.render_template("maintenance_dashboard", {})).appendTo(page.body);

    frappe.call({
        method: "asset_management.asset_management.page.maintenance_dashboard.maintenance_dashboard.get_dashboard_data",
        callback: function(r) {
            if (!r.message) return;
            const data = r.message;

            $("#total_open").text(data.total_open);
            $("#total_critical").text(data.total_critical);

            new frappe.Chart("#status_pie_chart", {
                title: "Requests by Status",
                data: {
                    labels: data.pie_chart.labels,
                    datasets: [{ values: data.pie_chart.values }]
                },
                type: 'pie',
                height: 250
            });
        }
    });
};

File: asset_management/page/maintenance_dashboard/maintenance_dashboard.py
import frappe

@frappe.whitelist()
def get_dashboard_data():
    """Return dashboard metrics for Maintenance Requests."""
    data = {}

    # Total Open Requests
    data["total_open"] = frappe.db.count("Asset Maintenance Request", {"status": "Open"})

    # Total Critical Requests
    data["total_critical"] = frappe.db.count("Asset Maintenance Request", {"priority": "Critical"})

    # Pie chart data (group by status)
    status_data = frappe.db.get_all(
        "Asset Maintenance Request",
        fields=["status", "count(name) as total"],
        group_by="status",
        order_by="total desc"
    )

    data["pie_chart"] = {
        "labels": [d.status for d in status_data],
        "values": [d.total for d in status_data]
    }

    return data
